### PR TITLE
Elasticsearch export

### PR DESCRIPTION
**Pull request by Nishan8583**
Hello

I have an export function for Elasticsearch.  The export file is in exporters/es/es.go. Changes on others files were made accordingly. ELK is used a lot by security engineers and I had do some hacky stuff to export logs from OpenVAS to Elasticsearch in the past.  So I hope adding a export mechanism for Elasticsearch will help this project.

Note: make was not working from dev branch, so had to manually build it for testing.

### CODE REVIEW

**Comment by forgedhallpass** [comment]
Hello @Nishan8583 

As a first step, please change the target branch to "dev".

**Comment by Nishan8583** [comment]
> Hello @Nishan8583
> 
> As a first step, please change the target branch to "dev".
@forgedhallpass I just changed to dev.

File: v2/internal/runner/runner.go [hunk 1]
```diff
@@ -105,6 +106,28 @@ func New(options *types.Options) (*Runner, error) {
...
+				ESIP:              options.ESExport,
+				ESPort:            options.ESPort,
+				ESSSL:             options.ESSSL,
+				ESSSLVerificaiton: options.ESSSLVerificaiton,
+				ESUsername:        options.ESUsername,
+				ESPassword:        options.ESPassword,
+			}
+		} else {
+			reportingOptions = &reporting.Options{}
+			reportingOptions.ElasticsearchExporter = &es.Options{
```
Comment (forgedhallpass) [inline v2/internal/runner/runner.go:121]: it seems that you could extract the assignment out of the condition without duplicating it

File: v2/cmd/nuclei/main.go [hunk 2]
```diff
@@ -90,6 +90,13 @@ on extensive configurability, massive extensibility and ease of use.`)
 
 		flagSet.StringVarP(&options.DiskExportDirectory, "markdown-export", "me", "", "directory to export results in markdown format"),
 		flagSet.StringVarP(&options.SarifExport, "sarif-export", "se", "", "file to export results in SARIF format"),
+
+		flagSet.StringVar(&options.ESExport, "es-ip", "", "the ip address for elasticsearch . Ex: 127.0.0.1"),
+		flagSet.IntVar(&options.ESPort, "es-port", 9200, "the port number on which elasticsearch is running, by default 9200"),
```
Comment (forgedhallpass) [inline v2/cmd/nuclei/main.go:95]: Normally the default values provided ahould automatically shown when nuclei is invoked with `-h`. Could you please make sure that it does and if so remove the "defaults" from the description?

File: v2/internal/runner/runner.go (hunk 1 above)
Comment (Nishan8583) [reply v2/internal/runner/runner.go]: @forgedhallpass ok, I just pushed the changes.

File: v2/cmd/nuclei/main.go (hunk 2 above)
Comment (Nishan8583) [reply v2/cmd/nuclei/main.go]: I checked, the default is provided by `-h` option. I removed "defaults" from description and pushed it.

**Comment by Ice3man543** [comment]
Thanks for the PR @Nishan8583. Merging this into dev, after which I'll be making some changes to remove the CLI args for elasticsearch, and instead add support for using it from the reporting config.

**Comment by ehsandeep** [comment]
@Nishan8583 follow up changes - https://github.com/projectdiscovery/nuclei/pull/973, now instead of CLI, we can feed the ES config details in the reporting [configuration](https://github.com/projectdiscovery/nuclei/blob/dev/v2/cmd/nuclei/issue-tracker-config.yaml) file and can be used with `-report-config config.yaml`, thanks again for adding this support.

**Comment by Nishan8583** [comment]
> @Nishan8583 follow up changes - #973, now instead of CLI, we can feed the ES config details in the reporting [configuration](https://github.com/projectdiscovery/nuclei/blob/dev/v2/cmd/nuclei/issue-tracker-config.yaml) file and can be used with `-report-config config.yaml`, thanks again for adding this support.

@ehsandeep ok thanks.

**Pull request merged**